### ISSUE TITLE
``KJTAllToAllIndicesAwaitable`` related error:  torage size calculation overflowed with sizes=[8410156404468110876]

### ISSUE THREAD

**Issue by BackNode** [meta-pytorch/torchrec#1008]
Hi all,

I am hitting below ``KJTAllToAllIndicesAwaitable`` related error randomly during training stage. Any clue?

I am using torchrec 0.2, 1 node with 8 GPUs.

```
RuntimeError: Storage size calculation overflowed with sizes=[8410156404468110876]
or
RuntimeError: Trying to create tensor with negative dimension -3516029179369850022: [-3516029179369850022]
```
```
[1,7]<stderr>:  File "apps/ranking/hybrid_nn/modules/sharded_model_wrapper.py", line 303, in _step
[1,7]<stderr>:    loss, logits, labels, ids, rguids = self.pipelines[phase_type].progress(dataloader_iter)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/train_pipeline.py", line 506, in progress
[1,7]<stderr>:    (losses, output) = cast(Tuple[torch.Tensor, Out], self._model(batch_i))
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/model_parallel.py", line 276, in forward
[1,7]<stderr>:    return self._dmp_wrapped_module(*args, **kwargs)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 1008, in forward
[1,7]<stderr>:    output = self._run_ddp_forward(*inputs, **kwargs)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 969, in _run_ddp_forward
[1,7]<stderr>:    return module_to_run(*inputs[0], **kwargs[0])
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "apps/ranking/hybrid_nn/modules/sharded_model_wrapper.py", line 67, in forward
[1,7]<stderr>:    logits = self.model(features).squeeze()
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "apps/ranking/hybrid_nn/models/hybrid_model.py", line 78, in forward
[1,7]<stderr>:    sparse_outputs = self.sparse(sparse_kjt)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "core/nn/sparse.py", line 136, in forward
[1,7]<stderr>:    sparse_features: KeyedTensor = self.embedding_bag_collection(features)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
[1,7]<stderr>:    return forward_call(*input, **kwargs)
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/types.py", line 432, in forward
[1,7]<stderr>:    dist_input = self.input_dist(ctx, *input, **kwargs).wait()
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/embeddingbag.py", line 332, in input_dist
[1,7]<stderr>:    awaitables.append(all2all_lengths.wait())
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/types.py", line 110, in wait
[1,7]<stderr>:    ret: W = self._wait_impl()
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/embedding_sharding.py", line 119, in _wait_impl
[1,7]<stderr>:    id_list_features_awaitable=self._id_list_features_awaitable.wait()
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/types.py", line 110, in wait
[1,7]<stderr>:    ret: W = self._wait_impl()
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/dist_data.py", line 417, in _wait_impl
[1,7]<stderr>:    ret = KJTAllToAllIndicesAwaitable(
[1,7]<stderr>:  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torchrec/distributed/dist_data.py", line 188, in __init__
[1,7]<stderr>:    out_values = torch.empty(
[1,7]<stderr>:RuntimeError: Storage size calculation overflowed with sizes=[8410156404468110876]
```

**Comment by YLGH** [comment]
Haven't really seen this issue before, but looks like some kind of memory corruption - could you help with a small reproduction for us to investigate?

Also, not sure if it would help, but could you try disabling the training_pipeline

cc @joshuadeng for any additional insights

**Comment by joshuadeng** [comment]
`KJTAllToAllIndicesAwaitable` is from an older version of torchrec, which we have refactored. Can you try reproducing this with a newer version of torchrec?

**Comment by BackNode** [comment]
Thanks for checking.

@YLGH for disabling the training_pipeline, do you mean call ``forward, backward, optimize`` in user code, instead of using ``TrainPipelineSparseDist``?

@joshuadeng I am working on upgrading to TorchRec 0.3.2 meanwhile.

**Comment by BackNode** [comment]
The issue is gone after I upgrade to 0.3.2. Root cause: batch_size are different per rank at the end of loop, which will hit this exception in v0.2.0, while resolved in v0.3.2.

Thanks for your help, @YLGH , @joshuadeng !
